### ISSUE TITLE
Support for `std::unique_ptr` member properties?

### ISSUE THREAD

**Issue by taketwo** [rttrorg/rttr#39]
Hi, sorry, it's me again :) I was wondering if RTTR supports `std::unique_ptr` member properties. From the documentation of [`wrapper_mapper`](http://www.rttr.org/doc/master/structrttr_1_1wrapper__mapper.html) I conclude that they are supported. However, when I try to register a member field of type `std::unique_ptr`, I get the following [compilation error](http://pastebin.com/df24dSHP), which seems to originate in the fact that unique pointers are not copy-constructible. So what is the status actually?

```cpp
#include <rttr/registration>

struct A
{
  std::unique_ptr<int> p;
};

RTTR_REGISTRATION
{
  using namespace rttr;
  registration::class_<A>("A")
    .property_readonly("p", &A::p);
}
```

Compiler: GCC 5.4 and Clang 3.8.

**Comment by acki-m** [comment]
You are correct, `unique_ptr` is not copy-constructible. Thats why you have to use a policy to register this property. For example [this policy](http://www.rttr.org/doc/master/structrttr_1_1policy_1_1prop.html#a6e551c5748cfaad0d8f1afc353817f78).
Unfortunately there is a problem, that the default registration does not have this policy, that's why you get the compile error, even when you use the policy...That's a bug.

The ` wrapper_mapper ` is used inside a variant and in the instance class. For instance, you can set a property of an object, even when the instance is inside a wrapper. Take a look at the example:
```
std::shared_ptr<base_class> base_obj = std::make_shared<derived_class>();

type t = type::get(base_obj);
// will look up the whole class hierachy of 'base_class' and 'derived_class' for a property named 'visibility'
property prop = t.get_wrapped_type().get_property("visibility");

// it's possible to invoke the property, from every arbitrary class hierarchy level,
// it also doesn't matter, when its wrapped in a shared_ptr
prop.set_value(base_obj, false); 
```

You can put `base_obj` even inside a variant and invoke the setter with the variant, the property will be set.

**Comment by taketwo** [comment]
Thanks for the clarification.
